File: project/module-core/src/main/java/cn/fd/ratziel/core/exception/ArgumentNotFoundException.java
package cn.fd.ratziel.core.exception;

/**
 * ArgumentNotFoundException
 *
 * @author TheFloodDragon
 * @since 2024/8/16 18:28
 */
public final class ArgumentNotFoundException extends Exception {

    private final Class<?> type;

    public ArgumentNotFoundException(Class<?> type) {
        super("Cannot found the argument: " + type.getName() + " !");
        this.type = type;
    }

    public Class<?> getMissingType() {
        return type;
    }

}

File: project/module-core/src/main/kotlin/cn/fd/ratziel/core/contextual/ArgumentNotFoundException.kt
package cn.fd.ratziel.core.contextual

/**
 * ArgumentNotFoundException
 *
 * @author TheFloodDragon
 * @since 2024/8/16 18:28
 */
class ArgumentNotFoundException(val missingType: Class<*>) : Exception("Cannot found the argument: " + missingType.getName() + " !")
